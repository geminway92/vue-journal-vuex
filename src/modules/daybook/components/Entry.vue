<template>
  <div class="entry-container mb-3 pointer p-2"
       @click="$router.push({ name: 'entry', params: { id: 10} })">
      
      <!-- TÃ­tulo -->
      <div class="entry-title d-flex">
          <span class="text-success fs-5 fw-bold">15</span>
          <span class="mx-1 fs-5">Julio</span>
          <span class="mx-2 fw-light">2021, Jueves</span>
      </div>
      
      <div class="entry-description">
        Amet sint duis dolor velit dolore. Voluptate quis nisi magna eu. In nostrud incididunt commodo cillum laboris laborum ea. Mollit ad ex fugiat in labore proident nostrud laborum deserunt consequat deserunt. Minim culpa laborum proident sint consequat voluptate irure nostrud sit. Sunt aliquip veniam labore Lorem. Non incididunt incididunt Lorem eiusmod deserunt ex irure nulla labore veniam Lorem sint.
      </div>
  </div>
</template>

<script>
export default {

}
</script>

<style lang="scss">
.entry-container {
    border-bottom: 1px solid #2c3e50;
    transition: 0.2s all ease-in;

      &:hover {
        background-color: lighten($color: grey, $amount: 45);
        transition: 0.2s all ease-in;
      }
}

.entry-description {
  font-size: 12px;
}
</style>