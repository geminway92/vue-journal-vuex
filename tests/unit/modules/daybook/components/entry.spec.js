import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry.vue";
import { journalState } from "../../../mocks-data/test-journal-state"

describe("Pruebas en el Entry Component", () => {
  // mockRouter
  const mockRouter = {
    push: jest.fn()
  };

  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[0]
    },
    global: {
      mocks: {
        $router: mockRouter

      }
    }
  });

    test("debe de hacer match con el snapchot", () => {
      expect(wrapper.html()).toMatchSnapshot()
    }),

    test("debe de redireccionar al hacer click en el entry-container", () => {

      const entryContainer = wrapper.find('.entry-container')
      entryContainer.trigger('click')

      expect( mockRouter.push ).toHaveBeenCalledWith(
        {   name: 'entry',
            params: {
            id: journalState.entries[0].id
        
          }
      
        }
      )

    });

    test("pruebas en las propiedades computadas", () => {
      // wrapper.vm. <------- ver las porpiedades computadas
      // day: 21
      // month: Septiembre
      // yearDay: '2021, Viernes'

      expect(wrapper.vm.day).toBe(1)
      expect(wrapper.vm.month).toBe('Septiembre')
      expect(wrapper.vm.yearDay).toBe('2021, MiÃ©rcoles')
    })
})
