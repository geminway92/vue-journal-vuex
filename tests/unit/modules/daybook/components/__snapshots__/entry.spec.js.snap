// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Entry Component debe de hacer match con el snapchot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Título -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">1</span><span class="mx-1 fs-5">Septiembre</span><span class="mx-2 fw-light">2021, Miércoles</span></div>
  <div class="entry-description">Hola mundo, es una prueba</div>
</div>
`;
